### PR TITLE
Avoid raising PT012 for simple `with` statements

### LINKED ISSUE #6054: Is it possible to avoid raising `PT012` if `with pytest.raises` contains a `with` statement with a single raise statement?
Is it possible to avoid raising `PT012` if `with pytest.raises` contains a `with` statement with a single raise statement? 

Example

```python
def test_start_run_does_not_suppress_error():
    with pytest.raises(Exception, match="foo"):
        with mlflow.start_run(...):
            raise Exception("foo")

    assert ...
```

References

- https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT012.md

### PR DESCRIPTION
**Pull request by harupy**
## Summary

Close #6054

## Test Plan

A new test case.

### CODE REVIEW

File: crates/ruff/src/rules/flake8_pytest_style/rules/raises.rs [hunk 1]
```diff
@@ -83,11 +83,29 @@ fn is_pytest_raises(func: &Expr, semantic: &SemanticModel) -> bool {
...
+            | Stmt::AsyncFor(_)
+            | Stmt::Match(_)
+            | Stmt::With(_)
+            | Stmt::AsyncWith(_)
+            | Stmt::If(_)
+            | Stmt::Try(_)
+            | Stmt::TryStar(_)
+    )
+}
+
```
Comment (harupy) [inline crates/ruff/src/rules/flake8_pytest_style/rules/raises.rs:103]: https://github.com/astral-sh/ruff/blob/da33c26238d359e9e7a4d0549fc25b1e5c1ce7fc/crates/ruff_python_formatter/src/statement/suite.rs#L145

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_pytest_style/rules/raises.rs:103]: I'll move it to `ruff_python_ast`...

**Comment by harupy** [comment]
https://github.com/apache/airflow/blob/f2108892e89085f695f8a3f52e076b39288497c6/tests/utils/test_task_group.py#L1091

```python
def test_decorator_unknown_args():
    """Test that unknown args passed to the decorator cause an error at parse time"""
    with pytest.raises(TypeError):

        @task_group_decorator(b=2)
        def tg():
            ...
```

Should a function be allowed?

**Comment by harupy** [comment]
@charliermarsh wdyt?

**Comment by jamesbeith** [comment]
Any chance we could get the prior behaviour (before this PR) available again, perhaps behind a some sort of `strict = True` setting?

**Pull request merged**